### PR TITLE
KAA-705 When security keys contain junk data endpoint is not able to recover

### PR DESCRIPTION
**Pull request by forGGe**
New pair will be generated if keys storage is corrupted.

### CODE REVIEW

File: client/client-multi/client-cpp/kaa/security/KeyUtils.hpp [hunk 1]
```diff
@@ -45,6 +45,15 @@ class KeyUtils {
...
     SessionKey generateSessionKey(std::size_t length);
 
+    /**
+     * Checks consistency of the key pair.
+     *
+     * @param[in] keys     Target key pair
+     * @retval    true     Keys are valid
+     * @retval    false    Keys are invalid
+     */
+    bool checkKeyPair(const KeyPair &keys);
```
Comment (ghost) [inline client/client-multi/client-cpp/kaa/security/KeyUtils.hpp:55]: Please, make it static.

File: client/client-multi/client-cpp/impl/KaaClient.cpp [hunk 2]
```diff
@@ -283,15 +283,24 @@ void KaaClient::initClientKeys()
 {
     std::string publicKeyLocation = context_.getProperties().getPublicKeyFileName();
     std::string privateKeyLocation = context_.getProperties().getPrivateKeyFileName();
+    KeyUtils utils;
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/KaaClient.cpp:286]: Please, use all functions from KeyUtils as static.

File: client/client-multi/client-cpp/impl/security/KeyUtils.cpp [hunk 3]
```diff
@@ -32,6 +32,30 @@ SessionKey KeyUtils::generateSessionKey(std::size_t length)
...
+            return result;
+
+        auto &priv = keys.getPrivateKey();
+        Botan::DataSource_Memory src(priv);
+        std::unique_ptr<Botan::Private_Key> ppriv{Botan::PKCS8::load_key(src, rng_)};
+        result = ppriv->check_key(rng_, strongCheck);
+
+        return result;
+    }
+    catch(...) {
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/security/KeyUtils.cpp]: Please, place catch block on the same line with enclosing brace.

**Comment by ghost** [comment]
Reviewed.

File: client/client-multi/client-cpp/impl/security/KeyUtils.cpp (hunk 3 above)
Comment (ghost) [inline client/client-multi/client-cpp/impl/security/KeyUtils.cpp]: Invalid

File: client/client-multi/client-cpp/impl/KaaClient.cpp (hunk 2 above)
Comment (ghost) [inline client/client-multi/client-cpp/impl/KaaClient.cpp:286]: Invalid

**Comment by ohamov** [comment]
LGTM

**Comment by ghost** [comment]
LGTM

**Pull request merged**